languages.toml

```diff
@@ -1224,3 +1224,17 @@ indent = { tab-width = 2, unit = "  " }
 [[grammar]]
 name = "nu"
 source = { git = "https://github.com/LhKipp/tree-sitter-nu", rev = "db4e990b78824c8abef3618e0f93b7fe1e8f4c0d" }
+
+[[language]]
+name = "vala"
+scope = "source.vala"
+injection-regex = "vala"
+file-types = ["vala", "vapi", "genie"]
+roots = [".git", "meson.build"]
```

Comment:
Git and Meson are not Vala-specific in any way so roots should probably be left empty. It seems quite common to have empty roots in this file.

Reply:
for vala project, it is very common using `meson.build`.

seems that helix will prefer to `.git` and then fallback to `root_markers` ?

Reply:
for vala project, it is very common using `meson.build`.

seems that helix will prefer to `.git` and then fallback to `root_markers` ?

Reply:
if it is that way, I think it is OK to remove `meson.build` from `root_markers`, and leave the use to config it, if they did need this.

Reply:
according to [wiki.gnome.org/Projects/Vala/Tools](https://wiki.gnome.org/Projects/Vala/Tools)

> Meson - a front end to the Ninja build system with Vala support (The main build system for Vala)

meson is not the only, but is the `main`

Reply:
just remove both now, `roots` is by default empty